oauth2-platform-api/src/main/java/com/intuit/oauth2/http/HttpRequestClient.java

```diff
@@ -53,233 +52,215 @@
-        //create oauth consumer using tokens
-	    OAuthConsumer consumer = new CommonsHttpOAuthConsumer(migrationRequest.getConsumerKey(), migrationRequest.getConsumerSecret());
+
+	private static final int CONNECTION_TIMEOUT = 10000;
+	private static final int SOCKET_TIMEOUT = 30000;
+	private static final Logger logger = LoggerImpl.getInstance();
+	private final CloseableHttpClient client;
+
+	/**
+	 * Build the HttpClient
+	 */
+	public HttpRequestClient(ProxyConfig proxyConfig) {
+		RequestConfig config = RequestConfig.custom().setConnectTimeout(CONNECTION_TIMEOUT).setSocketTimeout(SOCKET_TIMEOUT).build();
+
+		//add default headers
+		List<BasicHeader> headers = new ArrayList<BasicHeader>();
+		headers.add(new BasicHeader(HttpHeaders.ACCEPT_ENCODING, "utf-8"));
+		headers.add(new BasicHeader(HttpHeaders.ACCEPT, "application/json"));
+		headers.add(new BasicHeader(HttpHeaders.USER_AGENT, "V3JavaSDK-OAuth2-" + PropertiesConfig.getInstance().getProperty("version")));
+
+		//build the client
+		Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create().register("https", prepareClientSSL()).register("http", new PlainConnectionSocketFactory()).build();
+		PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+		HttpClientBuilder hcBuilder = HttpClients.custom().setConnectionManager(cm).setDefaultRequestConfig(config).setDefaultHeaders(headers).setMaxConnPerRoute(10).setDefaultCredentialsProvider(setProxyAuthentication(proxyConfig));
+
+		// getting proxy from Config file.
+		HttpHost proxy = getProxy(proxyConfig);
+
+		if (proxy != null) {
+			hcBuilder.setProxy(proxy);
+		}
+		client = hcBuilder.build();
+	}
+
+	public SSLConnectionSocketFactory prepareClientSSL() {
+		try {
+			KeyStore trustStore = null;
+			SSLContext sslContext = SSLContexts.custom().loadTrustMaterial(trustStore, new TrustSelfSignedStrategy()).build();
+
+			String tlsVersion = PropertiesConfig.getInstance().getProperty("TLS_VERSION");
+			SSLConnectionSocketFactory sslConnectionFactory = new SSLConnectionSocketFactory(sslContext, new String[]{tlsVersion}, null, new NoopHostnameVerifier());
+			return sslConnectionFactory;
+		} catch (Exception ex) {
+			logger.error("couldn't create httpClient!! {}", ex.getMessage(), ex);
+			return null;
+		}
+	}
+
+	/**
+	 * Method to set proxy authentication
+	 *
+	 * @return
+	 */
+	public CredentialsProvider setProxyAuthentication(ProxyConfig proxyConfig) {
+
+		if (proxyConfig == null) {
+			return null;
+		}
+		String username = proxyConfig.getUsername();
+		String password = proxyConfig.getPassword();
+
+		if (!username.isEmpty() && !password.isEmpty()) {
+			String host = proxyConfig.getHost();
+			String port = proxyConfig.getPort();
+			if (!host.isEmpty() && !port.isEmpty()) {
+				CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+				if (username.contains("/") || username.contains("\\")) {
```

Comment:
Instead of adding this check, let's add a new field called domain in the ProxyConfig and if domain is found lets use NTCredentials else use UsernamePasswordCredentials.